### ISSUE TITLE
How do I scale SOLVER.STEPS with SOLVER.MAX_ITER ?

### ISSUE THREAD

**Issue by PhilipMay** [facebookresearch/detectron2#1238]
[Base-RCNN-FPN.yaml](https://github.com/facebookresearch/detectron2/blob/master/configs/Base-RCNN-FPN.yaml) is trained in basic config with `STEPS`: (60000, 80000) and `MAX_ITER`: 90000

In [faster_rcnn_R_50_FPN_3x.yaml](https://github.com/facebookresearch/detectron2/blob/master/configs/COCO-Detection/faster_rcnn_R_50_FPN_3x.yaml) you scale `MAX_ITER` by a factor of 3 to 270000.
But `STEPS` is scaled to (210000, 250000) in the same config.

My question is: How do you calculate the new `STEPS` values when you scale up `MAX_ITER` by a factor?

My problem is that the propotions 60, 80 and 90 do not seem to fit to 210, 250 and 270.

**Background to my question:**
I want to change the `IMS_PER_BATCH` for faster_rcnn_R_50_FPN_3x.yaml to 4 (single GPU).
Now I have to scale up the `MAX_ITER`. But how do I scale up `STEPS`?

Thanks
Philip

**Comment by ppwwyyxx** [comment]
we follow the method described in https://arxiv.org/abs/1811.08883

**Comment by PhilipMay** [comment]
Hi @ppwwyyxx many thanks for the quick answer! :+1: 

> Following the strategy in the 2×schedule, we always reduce the learning rate by 10×in the last60k and last 20k iterations respectively, no matter how manytotal iterations (i.e.,  the reduced learning rates are alwaysrun for the same number of iterations).

So when you train with 2x (or more) schedule you reduce the last 60k and 20k iterations.

Can you please tell me how to scale this with the batch size?
My guess: When I have a batch size of 8 instead of 16 I sould increase this to "last 120k and 40k iterations". Do you agree with this?

If you want to I could add these informations to the doc by a PR. I think this could be added here: [Training & Evaluation in Command Line](https://detectron2.readthedocs.io/tutorials/getting_started.html#training-evaluation-in-command-line)

**Comment by ppwwyyxx** [comment]
It's up to you. We have already referred to a paper we recommend in that exact section in the docs.

**Comment by PhilipMay** [comment]
> 
> 
> It's up to you. We have already referred to a paper we recommend in that exact section in the docs.

Yes thanks. The papers do not say something on how to scale "the last iterations" with batch size. Can you please tell us something about this question?
